storybookjs/storybook
# Args: Update ArgType type definition

## What I did

Updated `ArgTypes` typescript definition to reflect what is documented in https://storybook.js.org/docs/react/api/arg-types

## Checklist for Contributors

### Testing

#### The changes in this PR are covered in the following automated tests:
- [X] typescript types

#### Manual testing

... I am embarassed to admit I couldn't figure out how to run typescript check on my computer. I'm hoping there's a CI pipeline.

### Documentation

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>
